modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java

```diff
@@ -277,44 +307,61 @@ public void testPersistedCorrelationConfigs() throws Exception {
-        }
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            String logLine;
+            while ((logLine = bufferedReader.readLine()) != null) {
+                log.info("***p1 logLine: " + logLine);
+                assertTrue(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
+            }
+        });
+
 
         serverConfigurationManager.restartGracefully();
         Thread.sleep(10000);
-
-        while ((logLine = bufferedReader.readLine()) != null) {
-            log.info("***p2 logLine: " + logLine);
-            if (logLine.contains("Started log handler")) {
-                break;
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            String logLine;
+            while ((logLine = bufferedReader.readLine()) != null) {
+                log.info("***p2 logLine: " + logLine);
+                if (logLine.contains("Started log handler")) {
+                    break;
+                }
             }
-        }
+        });
+
 
         resetAllLogs();
         InvokeTestAPI();
-        
-        while ((logLine = bufferedReader.readLine()) != null) {
-            log.info("***p3 logLine: " + logLine);
-            assertTrue(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
-        }
-        assertTrue(httpLog && methodCallsLog);
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            String logLine;
+            while ((logLine = bufferedReader.readLine()) != null) {
+                log.info("***p3 logLine: " + logLine);
+                assertTrue(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
+            }
+            assertTrue(httpLog && methodCallsLog);
+        });
+
 
         configureCorrelationLoggingComponent(new String[] { "http", "method-calls" }, false);
-        while ((logLine = bufferedReader.readLine()) != null) {
-            log.info("***p4 logLine: " + logLine);
-            assertTrue(isHTTPLogLine(logLine));
-        }
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            String logLine;
+            while ((logLine = bufferedReader.readLine()) != null) {
+                log.info("***p4 logLine: " + logLine);
+                assertTrue(isHTTPLogLine(logLine));
+            }
+        });
+
 
         // To check whehther no logs are printing after disabling
         resetAllLogs();
         InvokeTestAPI();
-        while ((logLine = bufferedReader.readLine()) != null) {
-            log.info("***p5 logLine: " + logLine);
-            assertFalse(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
-        }
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            String logLine;
+            while ((logLine = bufferedReader.readLine()) != null) {
+                log.info("***p5 logLine: " + logLine);
+                assertFalse(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
+            }
+        });
+
```

Comment:
remove unnecessary empty lines at other places as well

Reply:
Fixed in https://github.com/wso2/product-apim/pull/13088/commits/2c7b40a5f8ca5cab63ad8b5cb0d8bc511237cc5a